packages/roosterjs-editor-core/lib/editor/CorePlugin.ts

```diff
@@ -210,22 +217,38 @@ export default class CorePlugin implements EditorPlugin {
         }
     }
 
+    private onKeyPress(event: PluginKeyboardEvent) {
+        // If normalization was not possible before the keypress,
+        // check again after the keyboard event has been processed by browser native behaviour.
+        //
+        // This handles the case where the keyboard event that first inserts content happens when
+        // there is already content under the selection (e.g. Ctrl+a -> type new content).
+        if (!this.tryNormalizeTyping(event)) {
+            requestAnimationFrame(() => {

```

Comment:
So in most case, "findClosestElementAncestor(range.startContainer) == this.contentDiv" will be false, which means it will request an animation frame. This will happen to almost all key input. I think here we are trying to fix the Ctrl+A case, but should not impart all the normal input.

Reply:
You can check range.collapsed, and only do this when it is false.

Reply:
:+1:

Reply:
also just want to note that though it would schedule a raf, it wouldn't act within it, so it wouldn't be _that_ much of a hit.